{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Great, now that we discussed a little let's continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given that the current approach utilized by the authors lacks reproducibility, we will explore an alternative method by leveraging nf-core pipelines for data analysis.\n",
    "\n",
    "Please explain, how we will achieve reproducibility for the course  with this approach.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "specify parameters in a file with a constant run environment, constant tools"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You have successfully downloaded 2 of the fastq files we will use in our study.\n",
    "\n",
    "What is the next step if we want to first have a count table and check the quality of our fastq files? What is the pipeline called to do so?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "rna-seq"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analyze the 2 files using an nf-core pipeline.\n",
    "\n",
    "What does this pipeline do?\n",
    "\n",
    "Which are the main tools that will be used in the pipeline?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "HISAT2 for alignment if we follow the previous paper, STAR-Salmon as default"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As all other nf-core pipelines, the chosen pipeline takes in a samplesheet as input.\n",
    "\n",
    "Use Python and pandas to create the samplesheet for your 2 samples. Feel free to make use of the table you created earlier today.\n",
    "\n",
    "see part1 notebook\n",
    "\n",
    "Choose your sample names wisely, they must be the connection of the results to the metadata. If you can't find the sample in the metadata later, the analysis was useless."
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-09-30T12:59:31.874434Z",
     "start_time": "2025-09-30T12:57:05.694745Z"
    }
   },
   "source": [
    "# post here the command you used to run nf-core/rnaseq\n",
    "!nextflow run nf-core/rnaseq \\\n",
    "  --input sample_sheet.csv \\\n",
    "  --aligner hisat2 \\\n",
    "  --outdir results_rnaseq \\\n",
    "  -profile docker \\\n",
    "  --genome GRCm38 \\\n",
    " -c nextflow.config"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\u001B[1m\u001B[38;5;232m\u001B[48;5;43m N E X T F L O W \u001B[0;2m  ~  \u001B[mversion 25.04.7\u001B[m\r\n",
      "\u001B[K\r\n",
      "Launching\u001B[35m `https://github.com/nf-core/rnaseq` \u001B[0;2m[\u001B[0;1;36mmighty_banach\u001B[0;2m] DSL2 - \u001B[36mrevision: \u001B[0;36m9738a2df42 [master]\u001B[m\r\n",
      "\u001B[K\r\n",
      "\r\n",
      "------------------------------------------------------\r\n",
      "                                        ,--./,-.\r\n",
      "        ___     __   __   __   ___     /,-._.--~'\r\n",
      "  |\\ | |__  __ /  ` /  \\ |__) |__         }  {\r\n",
      "  | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,\r\n",
      "                                        `._,._,'\r\n",
      "  nf-core/rnaseq 3.21.0\r\n",
      "------------------------------------------------------\r\n",
      "\u001B[1mInput/output options\u001B[0m\r\n",
      "  \u001B[0;34minput              : \u001B[0;32msample_sheet.csv\u001B[0m\r\n",
      "  \u001B[0;34moutdir             : \u001B[0;32mresults_rnaseq\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mReference genome options\u001B[0m\r\n",
      "  \u001B[0;34mgenome             : \u001B[0;32mGRCm38\u001B[0m\r\n",
      "  \u001B[0;34mfasta              : \u001B[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001B[0m\r\n",
      "  \u001B[0;34mgtf                : \u001B[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001B[0m\r\n",
      "  \u001B[0;34mgene_bed           : \u001B[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.bed\u001B[0m\r\n",
      "  \u001B[0;34mstar_index         : \u001B[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/STARIndex/\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mUMI options\u001B[0m\r\n",
      "  \u001B[0;34mumi_discard_read   : \u001B[0;32m0\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mAlignment options\u001B[0m\r\n",
      "  \u001B[0;34maligner            : \u001B[0;32mhisat2\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mGeneric options\u001B[0m\r\n",
      "  \u001B[0;34mtrace_report_suffix: \u001B[0;32m2025-09-30_14-57-08\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mCore Nextflow options\u001B[0m\r\n",
      "  \u001B[0;34mrevision           : \u001B[0;32mmaster\u001B[0m\r\n",
      "  \u001B[0;34mrunName            : \u001B[0;32mmighty_banach\u001B[0m\r\n",
      "  \u001B[0;34mcontainerEngine    : \u001B[0;32mdocker\u001B[0m\r\n",
      "  \u001B[0;34mlaunchDir          : \u001B[0;32m/Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02\u001B[0m\r\n",
      "  \u001B[0;34mworkDir            : \u001B[0;32m/Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/work\u001B[0m\r\n",
      "  \u001B[0;34mprojectDir         : \u001B[0;32m/Users/patrick/.nextflow/assets/nf-core/rnaseq\u001B[0m\r\n",
      "  \u001B[0;34muserName           : \u001B[0;32mpatrick\u001B[0m\r\n",
      "  \u001B[0;34mprofile            : \u001B[0;32mdocker\u001B[0m\r\n",
      "  \u001B[0;34mconfigFiles        : \u001B[0;32m/Users/patrick/.nextflow/assets/nf-core/rnaseq/nextflow.config, /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/nextflow.config, /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/nextflow.config\u001B[0m\r\n",
      "\r\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\r\n",
      "-\u001B[2m----------------------------------------------------\u001B[0m-\r\n",
      "* The pipeline\r\n",
      "    https://doi.org/10.5281/zenodo.1400710\r\n",
      "\r\n",
      "* The nf-core framework\r\n",
      "    https://doi.org/10.1038/s41587-020-0439-x\r\n",
      "\r\n",
      "* Software dependencies\r\n",
      "    https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md\r\n",
      "\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001B[39m\u001B[K\r\n",
      "\u001B[8A\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ER_SETSTRANDEDNESS:\u001B[mFQ_LINT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m5\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "\u001B[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001B[39m\u001B[K\r\n",
      "\u001B[23A\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ER_SETSTRANDEDNESS:\u001B[mFQ_LINT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m23\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "\u001B[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001B[39m\u001B[K\r\n",
      "\u001B[23A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34mf6/972c9c\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 0 of 2\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "\u001B[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001B[39m\u001B[K\r\n",
      "\u001B[24A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34mf6/972c9c\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 0 of 2\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001B[K\r\n",
      "\u001B[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001B[39m\u001B[K\r\n",
      "\u001B[25A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34mf6/972c9c\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 0 of 2\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001B[K\r\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001B[K\r\n",
      "\u001B[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001B[39m\u001B[K\r\n",
      "\u001B[26A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34mf6/972c9c\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 1 of 2\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE (SRR23195516)'\u001B[K\r\n",
      "\u001B[K\r\n",
      "Caused by:\u001B[K\r\n",
      "  Process requirement exceeds available CPUs -- req: 12; avail: 11\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command executed:\u001B[K\r\n",
      "\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_1.fastq.gz ] && ln -s SRX19144486_SRR23195516_1.fastq.gz SRR23195516_trimmed_1.fastq.gz\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_2.fastq.gz ] && ln -s SRX19144486_SRR23195516_2.fastq.gz SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  trim_galore \\\u001B[K\r\n",
      "      --fastqc_args '-t 12' \\\u001B[K\r\n",
      "      --cores 8 \\\u001B[K\r\n",
      "      --paired \\\u001B[K\r\n",
      "      --gzip \\\u001B[K\r\n",
      "      SRR23195516_trimmed_1.fastq.gz \\\u001B[K\r\n",
      "      SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  \u001B[K\r\n",
      "  cat <<-END_VERSIONS > versions.yml\u001B[K\r\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE\":\u001B[K\r\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001B[K\r\n",
      "      cutadapt: $(cutadapt --version)\u001B[K\r\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001B[K\r\n",
      "  END_VERSIONS\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command exit status:\u001B[K\r\n",
      "  -\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command output:\u001B[K\r\n",
      "  (empty)\u001B[K\r\n",
      "\u001B[K\r\n",
      "Work dir:\u001B[K\r\n",
      "  /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/work/1e/a482c40f560b4aa9e22fbbefcbf7c3\u001B[K\r\n",
      "\u001B[K\r\n",
      "Container:\u001B[K\r\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001B[K\r\n",
      "\u001B[K\r\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[66A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34mf6/972c9c\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 1 of 2\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OOLS_TRIMGALORE:\u001B[mTRIMGALORE\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001B[K\r\n",
      "\u001B[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE (SRR23195516)'\u001B[K\r\n",
      "\u001B[K\r\n",
      "Caused by:\u001B[K\r\n",
      "  Process requirement exceeds available CPUs -- req: 12; avail: 11\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command executed:\u001B[K\r\n",
      "\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_1.fastq.gz ] && ln -s SRX19144486_SRR23195516_1.fastq.gz SRR23195516_trimmed_1.fastq.gz\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_2.fastq.gz ] && ln -s SRX19144486_SRR23195516_2.fastq.gz SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  trim_galore \\\u001B[K\r\n",
      "      --fastqc_args '-t 12' \\\u001B[K\r\n",
      "      --cores 8 \\\u001B[K\r\n",
      "      --paired \\\u001B[K\r\n",
      "      --gzip \\\u001B[K\r\n",
      "      SRR23195516_trimmed_1.fastq.gz \\\u001B[K\r\n",
      "      SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  \u001B[K\r\n",
      "  cat <<-END_VERSIONS > versions.yml\u001B[K\r\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE\":\u001B[K\r\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001B[K\r\n",
      "      cutadapt: $(cutadapt --version)\u001B[K\r\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001B[K\r\n",
      "  END_VERSIONS\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command exit status:\u001B[K\r\n",
      "  -\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command output:\u001B[K\r\n",
      "  (empty)\u001B[K\r\n",
      "\u001B[K\r\n",
      "Work dir:\u001B[K\r\n",
      "  /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/work/1e/a482c40f560b4aa9e22fbbefcbf7c3\u001B[K\r\n",
      "\u001B[K\r\n",
      "Container:\u001B[K\r\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001B[K\r\n",
      "\u001B[K\r\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[67A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34mf6/972c9c\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 1 of 2\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m1e/a482c4\u001B[0;2m] \u001B[0;2mNFC…E:\u001B[mTRIMGALORE\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001B[K\r\n",
      "\u001B[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE (SRR23195516)'\u001B[K\r\n",
      "\u001B[K\r\n",
      "Caused by:\u001B[K\r\n",
      "  Process requirement exceeds available CPUs -- req: 12; avail: 11\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command executed:\u001B[K\r\n",
      "\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_1.fastq.gz ] && ln -s SRX19144486_SRR23195516_1.fastq.gz SRR23195516_trimmed_1.fastq.gz\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_2.fastq.gz ] && ln -s SRX19144486_SRR23195516_2.fastq.gz SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  trim_galore \\\u001B[K\r\n",
      "      --fastqc_args '-t 12' \\\u001B[K\r\n",
      "      --cores 8 \\\u001B[K\r\n",
      "      --paired \\\u001B[K\r\n",
      "      --gzip \\\u001B[K\r\n",
      "      SRR23195516_trimmed_1.fastq.gz \\\u001B[K\r\n",
      "      SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  \u001B[K\r\n",
      "  cat <<-END_VERSIONS > versions.yml\u001B[K\r\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE\":\u001B[K\r\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001B[K\r\n",
      "      cutadapt: $(cutadapt --version)\u001B[K\r\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001B[K\r\n",
      "  END_VERSIONS\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command exit status:\u001B[K\r\n",
      "  -\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command output:\u001B[K\r\n",
      "  (empty)\u001B[K\r\n",
      "\u001B[K\r\n",
      "Work dir:\u001B[K\r\n",
      "  /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/work/1e/a482c40f560b4aa9e22fbbefcbf7c3\u001B[K\r\n",
      "\u001B[K\r\n",
      "Container:\u001B[K\r\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001B[K\r\n",
      "\u001B[K\r\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[67A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m4c/8403e6\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195511\u001B[2m)\u001B[m\u001B[2m |\u001B[m 2 of 2\u001B[32m ✔\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m1e/a482c4\u001B[0;2m] \u001B[0;2mNFC…E:\u001B[mTRIMGALORE\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001B[K\r\n",
      "-\u001B[0;35m[nf-core/rnaseq]\u001B[0;31m Pipeline completed with errors\u001B[0m-\u001B[K\r\n",
      "\u001B[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE (SRR23195516)'\u001B[K\r\n",
      "\u001B[K\r\n",
      "Caused by:\u001B[K\r\n",
      "  Process requirement exceeds available CPUs -- req: 12; avail: 11\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command executed:\u001B[K\r\n",
      "\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_1.fastq.gz ] && ln -s SRX19144486_SRR23195516_1.fastq.gz SRR23195516_trimmed_1.fastq.gz\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_2.fastq.gz ] && ln -s SRX19144486_SRR23195516_2.fastq.gz SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  trim_galore \\\u001B[K\r\n",
      "      --fastqc_args '-t 12' \\\u001B[K\r\n",
      "      --cores 8 \\\u001B[K\r\n",
      "      --paired \\\u001B[K\r\n",
      "      --gzip \\\u001B[K\r\n",
      "      SRR23195516_trimmed_1.fastq.gz \\\u001B[K\r\n",
      "      SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  \u001B[K\r\n",
      "  cat <<-END_VERSIONS > versions.yml\u001B[K\r\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE\":\u001B[K\r\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001B[K\r\n",
      "      cutadapt: $(cutadapt --version)\u001B[K\r\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001B[K\r\n",
      "  END_VERSIONS\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command exit status:\u001B[K\r\n",
      "  -\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command output:\u001B[K\r\n",
      "  (empty)\u001B[K\r\n",
      "\u001B[K\r\n",
      "Work dir:\u001B[K\r\n",
      "  /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/work/1e/a482c40f560b4aa9e22fbbefcbf7c3\u001B[K\r\n",
      "\u001B[K\r\n",
      "Container:\u001B[K\r\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001B[K\r\n",
      "\u001B[K\r\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[68A\r\n",
      "\u001B[2mexecutor >  local (2)\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:PREPARE_GENOME:\u001B[mGTF_FILTER -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…OME:\u001B[mMAKE_TRANSCRIPTS_FASTA -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ENOME:\u001B[mCUSTOM_GETCHROMSIZES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…:\u001B[mHISAT2_EXTRACTSPLICESITES -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…REPARE_GENOME:\u001B[mHISAT2_BUILD -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ASEQ:RNASEQ:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_SETSTRANDEDNESS:\u001B[mCAT_FASTQ -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m4c/8403e6\u001B[0;2m] \u001B[0;2mNFC…NESS:\u001B[mFQ_LINT\u001B[33;2m (\u001B[0;33mSRR23195511\u001B[2m)\u001B[m\u001B[2m |\u001B[m 2 of 2\u001B[32m ✔\u001B[m\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…UMITOOLS_TRIMGALORE:\u001B[mFASTQC\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m1e/a482c4\u001B[0;2m] \u001B[0;2mNFC…E:\u001B[mTRIMGALORE\u001B[33;2m (\u001B[0;33mSRR23195516\u001B[2m)\u001B[m\u001B[2m |\u001B[m 0 of 1\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ESS:\u001B[mFQ_LINT_AFTER_TRIMMING -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mFQ_SUBSAMPLE -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…PLE_FQ_SALMON:\u001B[mSALMON_QUANT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…_ALIGN_HISAT2:\u001B[mHISAT2_ALIGN -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…ATS_SAMTOOLS:\u001B[mSAMTOOLS_SORT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_INDEX -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…TS_SAMTOOLS:\u001B[mSAMTOOLS_STATS -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_FLAGSTAT -\u001B[K\r\n",
      "\u001B[2m[\u001B[0;34m-        \u001B[0;2m] \u001B[0;2mNFC…SAMTOOLS:\u001B[mSAMTOOLS_IDXSTATS -\u001B[K\r\n",
      "\u001B[3;2mPlus \u001B[1m24\u001B[0;3;2m more processes waiting for tasks…\u001B[m\u001B[K\r\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001B[K\r\n",
      "-\u001B[0;35m[nf-core/rnaseq]\u001B[0;31m Pipeline completed with errors\u001B[0m-\u001B[K\r\n",
      "\u001B[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE (SRR23195516)'\u001B[K\r\n",
      "\u001B[K\r\n",
      "Caused by:\u001B[K\r\n",
      "  Process requirement exceeds available CPUs -- req: 12; avail: 11\u001B[K\r\n",
      "\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command executed:\u001B[K\r\n",
      "\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_1.fastq.gz ] && ln -s SRX19144486_SRR23195516_1.fastq.gz SRR23195516_trimmed_1.fastq.gz\u001B[K\r\n",
      "  [ ! -f  SRR23195516_trimmed_2.fastq.gz ] && ln -s SRX19144486_SRR23195516_2.fastq.gz SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  trim_galore \\\u001B[K\r\n",
      "      --fastqc_args '-t 12' \\\u001B[K\r\n",
      "      --cores 8 \\\u001B[K\r\n",
      "      --paired \\\u001B[K\r\n",
      "      --gzip \\\u001B[K\r\n",
      "      SRR23195516_trimmed_1.fastq.gz \\\u001B[K\r\n",
      "      SRR23195516_trimmed_2.fastq.gz\u001B[K\r\n",
      "  \u001B[K\r\n",
      "  cat <<-END_VERSIONS > versions.yml\u001B[K\r\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE\":\u001B[K\r\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001B[K\r\n",
      "      cutadapt: $(cutadapt --version)\u001B[K\r\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001B[K\r\n",
      "  END_VERSIONS\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command exit status:\u001B[K\r\n",
      "  -\u001B[K\r\n",
      "\u001B[K\r\n",
      "Command output:\u001B[K\r\n",
      "  (empty)\u001B[K\r\n",
      "\u001B[K\r\n",
      "Work dir:\u001B[K\r\n",
      "  /Users/patrick/Studium_Local/3.Sem/computational-workflows-2025/notebooks/day_02/work/1e/a482c40f560b4aa9e22fbbefcbf7c3\u001B[K\r\n",
      "\u001B[K\r\n",
      "Container:\u001B[K\r\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001B[K\r\n",
      "\u001B[K\r\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "\u001B[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001B[K\r\n",
      "\u001B[K\r\n",
      " -- Check '.nextflow.log' file for details\u001B[39m\u001B[K\r\n",
      "Waiting for file transfers to complete (2 files)\r\n",
      "ERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\r\n",
      "\r\n",
      " -- Check '.nextflow.log' file for details\r\n",
      "\r\n"
     ]
    }
   ],
   "execution_count": 2
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain all the parameters you set and why you set them in this way.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "use docker profile, use hisat2 as in paper, genome is ensembl id of mouse reference genome, -c nextflow.config was an attempt to limit system resources as this workflow asks for 36GB memory but it keeps erroring out regardless"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browsing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How did the pipeline perform?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "it did not due to system limitations"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain the quality control steps. Are you happy with the quality and why. If not, why not.\n",
    "Please give additional information on : \n",
    "- ribosomal rRNA\n",
    "- Duplication\n",
    "- GC content\n",
    "\n",
    "What are the possible steps that could lead to poorer results?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Would you exclude any samples? If yes, which and why?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What would you now do to continue the experiment? What are the scientists trying to figure out? Which packages on R or python would you use?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "deseq2"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "work",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
